### PR TITLE
let user know about deprecated groups/properties for duplicate error

### PR DESCRIPTION
**Pull request by mkangia**
## Product Description

On Data dictionary,
Deprecated groups are hidden on the page and can block users from creating a group with a name that is already used by a deprecated group.
Minor change to update the error message to let user know about revealing deprecated groups in case they don't see the duplicate.

Same change being done for error message for properties.

-----
Before
image

-----
After
image

## Technical Summary

https://dimagi.atlassian.net/browse/SC-3744

### Safety story

Tested locally.

### QA Plan

No QA planned

### Rollback instructions

- [X] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [X] Risk label is set correctly
- [X] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Comment by zandre-eng** [comment]
@mkangia Since we have access to all the groups in JS, would it be possible for us to check whether the duplicate group name is deprecated? That way we can show an error message saying "A deprecated case property group with this name already exists. Please click..." if it is and just the normal error message if it isn't.

If the LOE on the above is too high however, then the current implementation looks good. Should we apply the same error message for deprecated case property names?

**Comment by mkangia** [comment]
Hey @zandre-eng 

ya, we could consider something better later. Considering the LoE of this would block the release and that the data dictionary is already undergoing many changes for which this might need to be adjusted again, I am keen to just keep it simple for now.

> Should we apply the same error message for deprecated case property names?

We can. let me see if I can include that in this PR itself.

**Comment by zandre-eng** [comment]
> ya, we could consider something better later. Considering the LoE of this would block the release and that the data dictionary is already undergoing many changes for which this might need to be adjusted again, I am keen to just keep it simple for now.

That sounds good, thanks Manish.

> We can. let me see if I can include that in this PR itself.

Happy to approve once it's been confirmed whether this change will happen in this PR or a separate one.

**Comment by mkangia** [comment]
Thanks for the suggestion @zandre-eng 
Added.

**Review by zandre-eng — APPROVED** [review]
Thanks for the change @mkangia, PR looks good.

**Pull request merged**
